File: dotnet/coverage-tests/test/Playwright/FirefoxBuilder.cs
using Microsoft.Playwright;

namespace Applitools.Generated.Playwright.Tests
{
    public class FirefoxBuilder : IPlaywrightBuilder
    {
        public IPage Build(IPlaywright playwright, bool headless, bool legacy, bool executionGrid)
        {
            IBrowser browser = playwright.Firefox.LaunchAsync(new BrowserTypeLaunchOptions
                {
                    Headless = headless,
                    IgnoreDefaultArgs = headless ? new[] { "--hide-scrollbars" } : null
                }
            ).GetAwaiter().GetResult();
            return browser.NewPageAsync().GetAwaiter().GetResult();
        }
    }
}

File: java/eyes.playwright.java/src/test/java/coverage/drivers/playwright/FirefoxBuilder.java
package coverage.drivers.playwright;

import com.microsoft.playwright.Browser;
import com.microsoft.playwright.BrowserType;
import com.microsoft.playwright.Page;
import com.microsoft.playwright.Playwright;

import java.net.MalformedURLException;
import java.util.Arrays;

public class FirefoxBuilder implements PlaywrightBuilder {
    @Override
    public Page build(Playwright playwright, boolean headless, boolean legacy, boolean executionGrid) throws MalformedURLException {
        Browser browser = playwright.firefox().launch(new BrowserType.LaunchOptions()
                .setHeadless(headless)
                .setIgnoreDefaultArgs(headless ? Arrays.asList("--hide-scrollbars") : null)
        );
        return browser.newPage();
    }
}
